### DISCUSSION TITLE
How detailed do we want the parsers to be?

### DISCUSSION THREAD

**Discussion by guidoism** [ArchiveBox/ArchiveBox#1394] [General]
I just downloaded my user data from TikTok and would like to import my favorites (and maybe my likes). It's a simple enough file to parse myself and pipe into `archivebox add` but I'm wondering what sort of granularity we want the parsers to be at. I see there are some specific ones for places like pinboard and some generic ones, including the newly added `jsonl`. 

So, do we want parsers for exported data from places like TikTok (I could easily contribute this to the project) or do we want to keep archivebox mostly generic?

**Comment by pirate** [comment]
Mostly generic, I don't want to have to support too many custom parsers.

I'd prefer people use other tools to export Tiktok stuff, then use tools like `jq`/`awk`/`sed`/etc. to manipulate it into a generic format that archivebox can ingest (UNIX philosophy style).

If you want to add any features to our existing parsers for a particular use-case like this one, I'd be happy to consider them, but I'd like to keep bespoke one-off parsers to a minimum for now.
